src/middlewared/middlewared/plugins/multipath.py

```diff
@@ -56,86 +52,36 @@ def query(self, filters, options):
               }
             ]
         """
+        return filter_list(self.__get_multipaths(), filters=filters or [], options=options or {})
 
-        if osc.IS_LINUX:
-            return []
-
-        multipaths = self.__get_multipaths()
-
-        items = []
-        for mp in multipaths:
+    def __get_multipaths(self):
+        doc = ET.fromstring(sysctl.filter("kern.geom.confxml")[0].value)
```

Comment:
We may want to leave a comment somewhere of why xml.etree and not lxml for anyone with bright ideas in the future.

Reply:
Kind of ashamed I didn't do that initially. Thanks for pointing it out. I've added a comment.